engine/src/main/java/org/destinationsol/game/FactionInfo.java

```diff
@@ -40,32 +34,26 @@ public FactionInfo() {
     }
 
     private void createFactionList() {
-        String[] moduleList = getModuleList();
-        for(String module: moduleList) {
-            String path =  module + ":factions";
-            System.out.println(path.equals("case:factions"));
-            System.out.println("core:factions");
-            Json factionJSON = Assets.getJson("core:factions");
-            for(int n = 0; n < factionJSON.getJsonValue().getJSONArray("factions").length(); n++) {
-                factionName.add(factionJSON.getJsonValue().getJSONArray("factions").getJSONObject(n).getString("name").replace("\"", ""));
-                factionColor.add(factionJSON.getJsonValue().getJSONArray("factions").getJSONObject(n).getString("color").replace("\"", ""));
-                factionDisposition.add(factionJSON.getJsonValue().getJSONArray("factions").getJSONObject(n).getInt("disposition"));
+        for (String module : getModuleList()) {
+            String path = module.replaceAll("[[, ]]", "") + ":factions";
+            Json factionJson = Assets.getJson(path);
+            for (int n = 0; n < factionJson.getJsonValue().getJSONArray("factions").length(); n++) {
+                factionName.add(factionJson.getJsonValue().getJSONArray("factions").getJSONObject(n).getString("name").replace("\"", ""));
+                factionColor.add(factionJson.getJsonValue().getJSONArray("factions").getJSONObject(n).getString("color").replace("\"", ""));
+                factionDisposition.add(factionJson.getJsonValue().getJSONArray("factions").getJSONObject(n).getInt("disposition"));
             }
         }
     }
 
-    private static String[] getModuleList() {
-        File file = new File("modules/");
-        String[] moduleList = file.list(new FilenameFilter() {
-            @Override
-            public boolean accept(File current, String name) {
-                return new File(current, name).isDirectory();
-            }
-        });
+    private static ArrayList<String> getModuleList() {
```

Comment:
Why do you get a list of all json files ending in "factions", then removing the "factions" from it and adding it again?

Reply:
Because I'm dumb and stopped thinking when writing that code

Reply:
This sounded like it was resolved - or? No need to think less of self over brain farts, we all get them :-)